test/widgets/action_sheet_test.dart

```diff
@@ -386,4 +390,97 @@ void main() {
       check(await Clipboard.getData('text/plain')).isNull();
     });
   });
+
+  group('StarButton', () {
+    Future<void> tapButton(WidgetTester tester) async {
+      // Locating button by the icon (as other tests here do) is complicated
+      // by the fact that starred messages are decorated with the same icon.
+      // Selecting by button class instead here:
+      await tester.ensureVisible(find.byType(StarButton, skipOffstage: false));
+      await tester.tap(find.byType(StarButton));
```

Comment:
Hmm, I see.

I think in my ideal version of this code, `StarButton` and its siblings would be private — they're part of the internal organization of `action_sheet.dart` but not part of its interface.

How about looking for the icon, but within the descendants of a `BottomSheet`? Can use `find.descendant`. If it's hard to make that work, this version is OK; but I suspect it'll be easy.